File: servidor-lora/src/infrastructure/services/validador-datos.service.ts
import { Injectable } from '@nestjs/common';

@Injectable()
export class ValidadorDatos {
  validarDatosAlerta(datos: any): boolean {
    return (
      typeof datos.devEUI === 'string' &&
      typeof datos.ubicacion?.lat === 'number' &&
      typeof datos.ubicacion?.lon === 'number' &&
      typeof datos.bateria === 'number'
    );
  } 
}

File: CODIGO/backend/backend_alert_csharp/backend_alert/Infrastructure/Services/ValidadorDatosService.cs
using Domain.Entities;

namespace Infrastructure.Services
{
    public class ValidadorDatosService
    {
        public bool ValidarDatosAlerta(Alerta alerta)
        {
            return !string.IsNullOrEmpty(alerta.DevEUI)
                && alerta.Lat != 0
                && alerta.Lon != 0
                && alerta.Bateria != 0;
        }
    }
}